### PR TITLE
Add some information on locale database to the ES docs

### PR DESCRIPTION
**Pull request by thecoop**
There needs to be some reference to locale database in the docs

### CODE REVIEW

File: docs/reference/setup/install.asciidoc [hunk 1]
```diff
@@ -98,6 +98,18 @@ the bundled JVM are treated as if they were within {es} itself.
...
+[[jdk-locale]]
+=== JDK locale database
+
+The locale database used by {es}, used to map from various date formats to
+the underlying date storage format, depends on the version of the JDK
+that {es} is running on. On JDK version 23 and above, {es} will use the
+_CLDR_ database. On JDK version 22 and below, {es} will use the _COMPAT_
+database. This may mean that the strings used for textual date formats,
+and the output of custom week-date formats, may change when moving from
+a previous JDK version to JDK 23 or above.
```
Comment (prdoyle) [inline docs/reference/setup/install.asciidoc:111]: Would it be appropriate to advise people specifically to avoid this? Specific things to avoid maybe?

Comment (thecoop) [reply docs/reference/setup/install.asciidoc]: Added some info

File: docs/reference/mapping/params/format.asciidoc [hunk 2]
```diff
@@ -31,8 +31,38 @@ down to the nearest day.
...
+
+* `B`, `E`, `G`, `O`, `a`, `v`, `z` of any length
+* `L`, `M`, `Q`, `q`, `c`, `e` of length 3 or greater
+* `Z` of length 4
+
+If the text format changes between Elasticsearch or JDK versions, it can cause significant problems
+with ingest, output, and re-indexing. It is recommended to always use numerical fields in custom date formats,
+which are not affected by locale information.
+
+There can also be differences in week-date calculations using the `Y`, `W`, and `w` field specifiers.
```
Comment (prdoyle) [inline docs/reference/mapping/params/format.asciidoc:54]: I love how explicit the list of field specifiers is.

Would it make sense to word this advice such that we could add these three stragglers to the bullet list above, to provide one-stop-shopping for users who want to ensure they avoid this problem? I could envision a user swiping the above list and pasting it into a Jira ticket, thereby missing these three.

File: docs/reference/mapping/params/format.asciidoc [hunk 3]
```diff
@@ -256,31 +286,37 @@ The following tables lists all the defaults ISO formats supported:
 `week_date` or `strict_week_date`::
 
     A formatter for a full date as four digit weekyear, two digit week of
-    weekyear, and one digit day of week: `xxxx-'W'ww-e`.
+    weekyear, and one digit day of week: `YYYY-'W'ww-e`.
+    This uses the ISO week-date definition.
```
Comment (prdoyle) [inline docs/reference/mapping/params/format.asciidoc:290]: 👍

**Review by prdoyle — APPROVED** [review]
Looks great!

File: docs/reference/mapping/params/format.asciidoc (hunk 2 above)
Comment (thecoop) [reply docs/reference/mapping/params/format.asciidoc:54]: Possibly, however the effects are quite different for each type, and I expect users to read the whole doc

**Pull request merged**